app/code/Magento/CatalogGraphQl/Model/Resolver/Products/DataProvider/ProductSearch.php

```diff
@@ -113,7 +113,7 @@ public function getList(
         $searchResults = $this->searchResultsFactory->create();
         $searchResults->setSearchCriteria($searchCriteriaForCollection);
         $searchResults->setItems($collection->getItems());
-        $searchResults->setTotalCount($searchResult->getTotalCount());
+        $searchResults->setTotalCount($collection->getSize());
```

Comment:
Hi sudheers-kensium,

Here why we have changed $searchResult->getTotalCount() to $collection->getSize() ?

Actually my concern is when we call getSize() on collection then it will execute one sql query to the database and it will take some time but in $searchResult is itself an object and all the data is present in that object. so, to execute $searchResult->getTotalCount() system will take negligible time where as $collection->getSize() will take some time.

I checked the performance of both code by adding below code in /vendor/magento/module-catalog-graph-ql/Model/Resolver/Products/DataProvider/ProductSearch.php file of "getList" method after "$this->collectionPostProcessor->process($collection, $attributes);" code and checked the response time of both getSize and getTotalCount method.
$time1 = microtime(true);
echo "Total Count: ".$collection->getSize()."<br/>";
$time2 = microtime(true);
echo "time taken in getSize is: ".($time2- $time1)."<br/>"; 

$time3 = microtime(true);
echo "Total Count: ".$searchResult->getTotalCount()."<br/>";
$time4 = microtime(true);
echo "time taken in getTotalCount is: ".($time4 - $time3); 
Response on my local :
Total Count: 32310
time taken in getSize is: 3.1487860679626
Total Count: 32310
time taken in getTotalCount is: 3.4093856811523E-5